### PR TITLE
[HOCS-6385] Match TO case data keys to DECS fields

### PR DESCRIPTION
**Pull request by jeff-horton-ho-sas**

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/cms/treatofficial/TreatOfficialMessageCaseData.java [hunk 1]
```diff
@@ -24,57 +26,33 @@ public TreatOfficialMessageCaseData(CaseDataTreatOfficialsRepository caseDataTre
...
+
+        return Stream.of(
+                        new CaseDataItem("OpenedDateTime", caseDataTreatOfficial.getOpenedDateTime()),
+                        new CaseDataItem("TypeId", caseDataTreatOfficial.getTypeId()),
+                        new CaseDataItem("ChannelReceived", "EMAIL"),
+                        new CaseDataItem("BusinessArea", caseDataTreatOfficial.getAllocatedToDeptId()),
+                        new CaseDataItem("DateReceived", IsoDateFormatter.format(caseDataTreatOfficial.getOpenedDateTime())),
+                        new CaseDataItem("ResponseDate", IsoDateFormatter.format(caseDataTreatOfficial.getResponseDate())),
+                        new CaseDataItem("TargetDate", IsoDateFormatter.format(caseDataTreatOfficial.getTargetDate())),
+                        new CaseDataItem("RejectionNote", caseDataTreatOfficial.getTxRejectNotes())
```
Comment (rramgolam) [inline src/main/java/uk/gov/digital/ho/hocs/cms/treatofficial/TreatOfficialMessageCaseData.java:39]: @jeff-horton-ho-sas I can see the letterTopic is added to the root schema, should it still be in the case-data blob?

Comment (rramgolam) [reply src/main/java/uk/gov/digital/ho/hocs/cms/treatofficial/TreatOfficialMessageCaseData.java:39]: Ok, is it required for extracts though?

Comment (rramgolam) [reply src/main/java/uk/gov/digital/ho/hocs/cms/treatofficial/TreatOfficialMessageCaseData.java:39]: For visibility, BA has confirmed that the letterTopic should appear in the extract and thus should be present in the TO case-data.

Comment (jeff-horton-ho-sas) [reply src/main/java/uk/gov/digital/ho/hocs/cms/treatofficial/TreatOfficialMessageCaseData.java:39]: primaryTopic  in the extract is set from the text of the primary topic linked to the case. That is why this ticket was complicated. Adding 'LetterTopic' as an arbitrary field in the case data blob will not cause it to show in the extract as 'LetterTopic' is not listed as a field to show in the extract for TO cases.

**Review by rramgolam — APPROVED** [review]

**Pull request merged**